src/Entity/EntityIdValue.php

```diff
@@ -123,11 +117,16 @@ public function getEntityId() {
 	 * @return array
 	 */
 	public function getArrayValue() {
-		return [
+		$array = [
 			'entity-type' => $this->entityId->getEntityType(),
-			'numeric-id' => $this->getNumericId(),
 			'id' => $this->entityId->getSerialization(),
 		];
+
+		if ( $this->entityId instanceof Int32EntityId ) {
```

Comment:
Sure. This is added here so this doesn't break for custom IDs that don't support numeric.